### PR TITLE
Re-impl CC_ASSERT & CC_ASSERTF

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: https://github.com/cocos/3d-tasks/issues/12144

### Changelog

* Impl CC_ASSERTF* family interfaces, print more information when assertion fails
* Replace CC_ASSERT(false) with CC_ABORT()

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by PatriceJiang** [comment]
@cocos-robot run test cases

File: native/cocos/base/Assertf.h [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+                char message[256] = {0};                                     \
+                snprintf(message, 256, fmt, ##__VA_ARGS__);                  \
+                __assert2(__FILE__, __LINE__, __PRETTY_FUNCTION__, message); \
+            } while (false)
+    #elif defined(__APPLE__)
+        #define CC_ASSERT_FORMAT(cond, fmt, ...)                                                \
+            do {                                                                                \
+                if (__builtin_expect(!(cond), 0)) {                                             \
+                    char message[256] = {0};                                                    \
+                    snprintf(message, 256, "CC_ASSERT(%s) failed. " fmt, #cond, ##__VA_ARGS__); \
```
Comment (dumganhar) [inline native/cocos/base/Assertf.h]:
```suggestion
                    snprintf(message, sizeof(message), "CC_ASSERT(%s) failed. " fmt, #cond, ##__VA_ARGS__); \
```

Comment (PatriceJiang) [reply native/cocos/base/Assertf.h]: fixed

File: native/cocos/base/Assertf.h [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                    __assert2(__FILE__, __LINE__, __PRETTY_FUNCTION__, message);                \
+                }                                                                               \
+            } while (false)
+        #define CC_ABORTF(fmt, ...)                                          \
+            do {                                                             \
+                char message[256] = {0};                                     \
+                snprintf(message, sizeof(message), fmt, ##__VA_ARGS__);      \
+                __assert2(__FILE__, __LINE__, __PRETTY_FUNCTION__, message); \
+            } while (false)
+    #elif defined(__APPLE__)
```
Comment (minggo) [inline native/cocos/base/Assertf.h]: I think it is better to use `CC_PLATFORM` here too.

File: native/cocos/scene/SubModel.cpp [hunk 3]
```diff
@@ -385,7 +385,7 @@ void SubModel::setInstancedAttribute(const ccstd::string &name, const float *val
                 case gfx::FormatType::SNORM:
                 case gfx::FormatType::UINT:
                 case gfx::FormatType::INT: {
-                    CC_ASSERT(false); // NOLINT
+                    CC_ABORT(); // NOLINT
```
Comment (minggo) [inline native/cocos/scene/SubModel.cpp]: Can remove `//NOLINT`.

File: native/cocos/base/Assertf.h [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    #define CC_ASSERT_ZERO(expr1)      _CC_ASSERT_CMP(expr1, ==, 0)
+    #define CC_ASSERT_NONZERO(expr1)   _CC_ASSERT_CMP(expr1, !=, 0)
+    /**
+     * @brief printf like assert
+     *
+     *  CC_ASSERTF(1==2, "value %d should not be equal to %d", 1, 2);
+     *  CC_ASSERTF_EQ(1, 3/3, "n equals to n");
+     *  CC_ASSERTF_NE(1, s, "not initial value");
+     *
+     */
```
Comment (minggo) [inline native/cocos/base/Assertf.h:138]: Please also add comment for `CC_ABORT()`

File: native/cocos/base/Assertf.h [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    #define CC_ASSERTF(cond, fmt, ...) CC_ASSERT_FORMAT(cond, fmt, ##__VA_ARGS__)
+    #define CC_ABORT()                 CC_ABORTF("CC_ABORT")
```
Comment (minggo) [inline native/cocos/base/Assertf.h]: It is meaningless to print `CC_ABORT`, please print more meaningful log.

File: native/cocos/base/Assertf.h (hunk 2 above)
Comment (PatriceJiang) [reply native/cocos/base/Assertf.h]: fixed

File: native/cocos/base/Assertf.h (hunk 4 above)
Comment (PatriceJiang) [reply native/cocos/base/Assertf.h:138]: fixed

File: native/cocos/base/Assertf.h (hunk 5 above)
Comment (PatriceJiang) [reply native/cocos/base/Assertf.h]: fixed

File: native/cocos/scene/SubModel.cpp (hunk 3 above)
Comment (PatriceJiang) [reply native/cocos/scene/SubModel.cpp]: fixed

**Review by minggo — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Pull request merged**